src/main/scala/mesosphere/marathon/state/Container.scala

```diff
@@ -33,43 +34,43 @@ case class Container(
 
 object Container {
 
-  object Empty extends Container
+  case class Mesos(volumes: Seq[Volume] = Seq.empty) extends Container
 
-  /**
-    * Docker-specific container parameters.
-    */
   case class Docker(
+    volumes: Seq[Volume] = Seq.empty,
     image: String = "",
-    network: Option[Mesos.ContainerInfo.DockerInfo.Network] = None,
-    portMappings: Option[Seq[Docker.PortMapping]] = None,
+    network: Option[ContainerInfo.DockerInfo.Network] = None,
+    pms: Seq[Docker.PortMapping] = Seq.empty,
```

Comment:
I guess this should be `Option[Seq[PortMapping]]` here as well.  The TaskBuilder differentiates between None and Seq.empty.